### PR TITLE
chore: staging api config

### PR DESCRIPTION
**Pull request by olizilla**
- update wrangler.toml with staging env
- update github action to publish staging api on merge to main

and out-of-band we have:
- web3.storage site created on Cloudflare
- nameservers updated to point web3.storage at Cloudflare
- DNSSEC enabled on web3.storage
- Scoped api token for editing workers on web3.storage added as `CF_API_TOKEN`

License: (Apache-2.0 AND MIT)
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
This is a reduced version of what is powering deploys in nft.storage. @hugomrdias can you give it a quick review pls.

File: .github/workflows/api.yml [hunk 1]
```diff
@@ -3,9 +3,15 @@ on:
   push:
     branches:
       - main
+    paths:
+      - 'packages/api/**'
+      - '.github/workflows/api.yml'
   pull_request:
```
Comment (alanshaw) [inline .github/workflows/api.yml:9]: Should we just remove this part and the `if` condition below?

Comment (olizilla) [reply .github/workflows/api.yml:9]: No, we want the test jon to run for pull requests that affect the api code. I could split these out into MOAR yamlz, if you'd prefer.

Comment (alanshaw) [reply .github/workflows/api.yml:9]: Ah, no, no more yamls - I missed the existing test job for some reason 🤦 .

**Review by alanshaw — APPROVED** [review]

**Pull request merged**